File: Python/242-validAnagram.py
class Solution:
    # @param {string} s
    # @param {string} t
    # @return {boolean}
    def isAnagramI(self, s, t):
        slst = list(s)
        tlst = list(t)
        slst.sort()
        tlst.sort()
        return slst == tlst

    def isAnagram(self, s, t):
        sdict = {}; tdict = {}
        for word in s:
            sdict[word] = sdict.get(word, 0) + 1
        for word in t:
            tdict[word] = tdict.get(word, 0) + 1
        for key in sdict:
            telem = tdict.pop(key, None)
            if telem is None:
               return False
            elif telem != sdict[key]:
                 return False
        return tdict == {}

File: C++/242-validAnagram.cpp
#include "include/common.h"

class Solution {
public:
    bool isAnagram(string s, string t) {
         if (s.length() != t.length()) {
             return false;
         };
         sort(s.begin(), s.end());
         sort(t.begin(), t.end());
         for(int i = 0; i < s.length(); i++) {
             if (s[i] != t[i]) {
                 return false;
             };
         };
         return true;
    }
};

int main(void) {
    Solution s;
    string test1 = "bdac", test2 = "abcd";
    cout << s.isAnagram(test1, test2) << endl; 
    test1 = "car", test2 = "rat";
    cout << s.isAnagram(test1, test2) << endl; 
    return 0;
};